### PR TITLE
Config: Implement better defaults

### PR DESCRIPTION
**Pull request by Geod24**
~Better reviewed commit-by-commit.
The large refactoring in main will be quite useful for the admin interface, where it's just a matter of adding an `if` at the top level.~

Stripped down to 1 commit...

### CODE REVIEW

File: source/agora/node/main.d [hunk 1]
```diff
@@ -75,31 +75,52 @@ private int main (string[] args)
             writefln("Config file '%s' succesfully parsed.", cmdln.config_path);
             return 0;
         }
+        runTask(() => runNode(config));
+    }
+    catch (ConfigException ex)
```
Comment (AndrejMitrovic) [inline source/agora/node/main.d]:
```suggestion
    catch (Exception ex)
```

File: source/agora/node/main.d [hunk 2]
```diff
@@ -75,31 +75,52 @@ private int main (string[] args)
             writefln("Config file '%s' succesfully parsed.", cmdln.config_path);
             return 0;
         }
+        runTask(() => runNode(config));
```
Comment (AndrejMitrovic) [inline source/agora/node/main.d]: Clever girl.

**Review by AndrejMitrovic — APPROVED** [review]

**Comment by Geod24** [comment]
Fixed

**Comment by AndrejMitrovic** [comment]
Is that a legit error in CircleCI?

**Comment by AndrejMitrovic** [comment]
> node-2_1_98233d7e6b9d | 2020-01-29T14:58:04.121257601Z Task terminated with uncaught exception: Configured quorum set is not considered valid by SCP

Hmm.. sounds like maybe the config wasn't parsed properly.

**Comment by Geod24** [comment]
Stripped down to 1 single commit, since this refactor discovered a few bugs.
Going to address everything independently and gradually submit the other commits.
Since the commit has already been approved, will self-merge once green.

**Pull request merged**